ICEBRG integration

Itay4 opened the pull request:
Fixes https://github.com/demisto/server/issues/6341

Spivaka commented on Integrations/integration-ICEBRG.yml:
```diff
@@ -0,0 +1,278 @@
+  id: ICEBRG
+  version: -1
+name: ICEBRG
+display: ICEBRG
```
we don't have all capital display names, names, etc.
its usually non capital in id and name, and first letter in display

Spivaka commented on Integrations/integration-ICEBRG.yml:
```diff
@@ -0,0 +1,278 @@
+  defaultvalue: https://events.icebrg.io/v2/query
+  type: 0
+  required: false
+- display: Server URL (e.g. https://192.168.0.1)
```
you have two identical display names names..

Spivaka commented on Integrations/integration-ICEBRG.yml:
```diff
@@ -0,0 +1,278 @@
+            });
+        }
+        var translator = {
+            contextPath: 'ICEBRG.' + path,
```
Icebrg instead

Spivaka commented on Integrations/integration-ICEBRG.yml:
```diff
@@ -0,0 +1,278 @@
+      description: When true, the service will include the service_traffic aggregation.
+        Default is false.
+    outputs:
+    - contextPath: ICEBRG.Events.QueryType
```
Icebrg not ICEBRG

orius123 commented on Integrations/integration-ICEBRG.yml:
```diff
@@ -0,0 +1,278 @@
+
+    switch (command) {
+        case 'test-module':
+            return 'ok';
```
Should add a real test

dorsha commented on Integrations/integration-ICEBRG.yml:
```diff
@@ -0,0 +1,278 @@
+  version: -1
+name: icebrg
+display: Icebrg
+category: Threat Intelligence
```
Network Security

Spivaka commented:
please fix comments from Eliya and Dorsha and merge

Itay4 commented:
### Overview:
ICEBRG is a network security product which is used through Demisto to get events and reports produced in ICEBRG for queries. 

### To set up ICEBRG to work with Demisto:
	1.	How to obtain API token (on ICEBRG):
		a. Go to ‘Settings > Profile Settings > Tokens’.
		b. Click ‘Create new token’.
		c. Enter description.	
		d. Click ‘Create’
		e. Record this token to use in the next steps.

### To set up the integration on Demisto:
For example:
	1	Go to ‘Settings > Integrations > Servers & Services’
	2	Locate ‘Icebrg’ by searching for it using the search box on the top of the page.
	3	Click ‘Add instance’ to create and configure a new integration. You should configure the following settings: Name: A textual name for the integration instance. Server URL for the search API: The URL of the search appliance.
		Server URL for the reports API: The URL of the reports appliance. ICEBRG token: ICEBRG API token. Fetch incidents: Select whether to automatically create Demisto incidents from this integration instance. Test: Checks correctness of URLs and token.

### Fetched data:
Fetching reports which contain more than one asset.
Events cannot be fetched.
Flittering by published date.
Fetching every 10 minutes.


### Top Use-cases:
Search events by query.
Get reports by UUID.

### Commands:
	1	Command name: icebrg-search-events
		Inputs:
		a. Query (mandatory) -  The query string or entity for which to search.
		b. Start date - The beginning of the temporal extent by which to restrict filter results, inclusive (in RFC3339 format).
		c. End date - The end of the temporal extent by which to restrict filter results, exclusive (in RFC3339 format).
		d. Order by - The event property by which to order results. Default: timestamp.
		e. Order - The order of results, either "asc" or "desc". Default: desc.
		f. Customer ID - The customer ID by which to restrict filter results. Default: user's account.
		g. History - When true, save this query in user's Query History and include up to the last 50 queries from user's Query History. Default: false.
		h. Service traffic - When true, the service will include the service_traffic aggregation. Default: false.
		Context output:
		a. Path: Icebrg.Events.QueryType      
                    Description: Query type
		b. Path: Icebrg.Events.Total      
                    Description: Total events
		c. Path: Icebrg.Events.OrderBy     
                    Description: Key to order events by
		d. Path: Icebrg.Events.Order     
                    Description: Order of the events
		e. Path: Icebrg.Events.Offset     
                    Description: Events offset
		f. Path: Icebrg.Events.History     
                   Description: History of events
		g. Path: Icebrg.Events.Limit     
                   Description: Limit number of events to show
		Raw output:
		{
    		"total": 135359505,
    		"offset": 0,
    		"limit": 100,
    		"order_by": "timestamp",
    		"query_type": "complex",
    		"events": [ ... ]
	        }
	2	Command name: icebrg-get-history
		Inputs:
								-
		Context output:
		a. Path: Icebrg.UserQueryHistory.Total     
                   Description: Total user queries
		b. Path: Icebrg.UserQueryHistory.Timestamp     
                    Description: Timestamp of user query
		c. Path: Icebrg.UserQueryHistory.Query     
                    Description: Called query
		d. Path: Icebrg.UserQueryHistory.QueryId     
                    Description: ID of query
		e. Path: Icebrg.UserQueryHistory.UserId     
                     Description: User ID
		Raw output:
		{
	 	"history": [{
        			"total": 3393897721,
        			"timestamp": "2017-03-30T20:56:11.556Z",
        			"query": "port = 80",
        			"id": "725be4f112f5b5ae9807b7130b2cea97"
   		 },
    		{
        			"total": 211313295,
        			"timestamp": "2017-03-30T17:44:35.748Z",
        			"query": "google.com",
        			"id": "655765009424c447765d06773e711dd3"
    		}],
    		"User_id": "f3259c9f-e54a-4e93-b71d-8e995a2cd96b"
	         }
	3	Command name: icebrg-saved-searches
		Inputs:
								-
		Context output:
		a. Path: Icebrg.SavedSearches.Tags     
                    Description: Query tags
		b. Path: Icebrg.SavedSearches.Description     
                    Description: Query description
		c. Path: Icebrg.SavedSearches.Title     
                    Description: Query title
		d. Path: Icebrg.SavedSearches.Timestamp     
                    Description: Query timestamp
		e. Path: Icebrg.SavedSearches.Query     
                    Description: Called query
		f. Path: Icebrg.SavedSearches.Id
		   Description: Query ID
		Raw output:
	        {
    		"saved_queries": [{
       			 "tags": [],
        			"description": "",
        			"title": "Test",
        			"timestamp": "2017-03-17T00:48:34.359Z",
        			"query": "ip='127.0.0.1'",
        			"id": "AVrZvNBGl0ZSNz2usg93"
    		}]
	         }
	4	Command name: icebrg-get-reports
		Inputs:
		a. Limit - The maximum number of records to return. The default is no limit.
		b. Offset - The number of records to skip past. The default is none.
		c. Sort by - The field to sort by created, updated, or published. The default is unsorted.
		d. Sort order - The sort order asc or desc. The default is asc if sort_by is provided.
		e. Account UUID - UUID of account to filter by.
		f. Archived - Archived status to filter by.
		g. Confidence - Confidence to filter by (low, moderate, high).	
		h. Risk - Risk to filter by (low, moderate, high).
		i. Search - Text string to search on title and summary.
		j. Status - Status to filter by.
		k. Published start - Published start date to filter by (inclusive), RFC3339 format.
		l. Published end - Published end date to filter by (exclusive), RFC3339 format.
		Context output:
		a. Path: Icebrg.Reports.Publishes.UserUuid     
                    Description: User UUID that published the report
		b. Path: Icebrg.Reports.Publishes.Published     
                    Description: Timestamp of published report
		c. Path: Icebrg.Reports.AssetCount     
                    Description: Asset count of report
		d. Path: Icebrg.Reports.IndicatorCount     
                    Description: Indicator count of report
		e. Path: Icebrg.Reports.Archived     
                    Description: True if archived, else false
		f. Path: Icebrg.Reports.Details     
                   Description: Report details
		g. Path: Icebrg.Reports.Summary     
                    Description: Report summary
		h. Path: Icebrg.Reports.Category
		    Description: Category of the report
		i.  Path: Icebrg.Reports.Confidence     
                    Description: Indicator count of report
		j.  Path: Icebrg.Reports.Archived     
                     Description: Confidence of report
		k. Path: Icebrg.Reports.Risk
		    Description: Risk of report
		l.  Path: Icebrg.Reports.Title     
                   Description: Report title
	         m. Path: Icebrg.Reports.Status
		    Description: Status of report
		n. Path: Icebrg.Reports.AccountUuid
		    Description: Account UUID of report
	        o.  Path: Icebrg.Reports.UpdatedUserUuid     
                  Description: User UUID that updated the report
		p. Path: Icebrg.Reports.CreatedUserUuid     
                    Description: User UUID that created the report
		q. Path: Icebrg.Reports.Updated
		    Description: Timestamp of report update
		r.  Path: Icebrg.Reports.Created     
                    Description: Timestamp of report creation
		s. Path: Icebrg.Reports.Uuid
		    Description: Report UUID
		Raw output:
	         {
  	          "reports": [{
    		 "publishes": [{
       			     "user_uuid": "b3fc3df4-d3cf-4202-971c-0dcfe7cccf42",
         			     "published": "2017-01-24T10:13:13.418Z"
      		   }],
      		 "asset_count": 1,
     		 "indicator_count": 5,
      		"archived": false,
      		"details": "On 21 January, ...",
      		"summary": "A host was infected with Cerber ransomware after opening a 			           malicious Word document received via email.",
      		"category": "Ransomware",
      		"confidence": "high",
      		"risk": "moderate",
      		"title": "Cerber Malware Infection",
      		"status": "published",
      		"account_uuid": "6bc3d2f1-af77-4236-a9db-17dacd06e4d9",
      		"updated_user_uuid": "b3fc3df4-d3cf-4202-971c-0dcfe7cccf42",
      		"created_user_uuid": "b3fc3df4-d3cf-4202-971c-0dcfe7cccf42",
      		"updated": "2017-01-24T10:12:32.534Z",
      		"created": "2017-01-24T07:25:36.363Z",
      		"uuid": "2d35734f-5b16-41ff-a482-b08a7c74202a"
   		}],
	        }
	5	Command name: icebrg-get-report-indicators
		Inputs:
		a. Report UUID (mandatory) - Report UUID to get indicator of
		Context output:
		a. Path: Icebrg.ReportIndicators.Type     
                     Description: Type of indicator
		b. Path: Icebrg.ReportIndicators.Indicators.Observable     
                    Description: Observable of Indicator
		Raw output:
	        {
    		"indicators" 		[{
        			"type": "ipv4",
        			"observable": "185.183.96.110"
    		}]
	        }
	6	Command name: icebrg-get-report-assets
		Inputs:
		a. Report UUID (mandatory) - Report UUID to get indicator of
		Context output:
		a. Path: Icebrg.ReportAssets.Asset     
                    Description: Assets of Report UUID
		Raw output:
	        {
   		"assets": [{
        			"asset" : "10.248.100.74"
    		}]
	         }

### Troubleshooting
This integration was integrated and tested with version 1.3 of ICEBRG.

orius123 commented:
@Itay4 Why did you delete script-http in this PR?

orius123 commented:
Moved to https://github.com/demisto/content/pull/1004

Spivaka commented:
also move documentation
